el: spread children parameter

crhallberg opened the pull request:
Spreading the children parameter opens up the opportunity for some more flexible and concise code. This can be done currently with an array, but this syntax prevents the need for an array when only one child is added.

```JS
el("p", { class: "lead" }, "text");

el(
  "ul", 
  {}, 
  el("li", {}, "item 1"),
  el("li", {}, "item 2"),
  el("li", {}, "item 3"),
  el("li", {}, "item 4"),
);
```

`.flat()` has been added to prevent this from being a breaking change.

crhallberg committed:
el: spread children parameter

Spreading the children parameter opens up the opportunity for some more flexible and concise code. This can be done currently with an array, but this syntax prevents the need for an array when only one child is added.

```JS
el("p", { class: "lead" }, "text");

el(
  "ul", 
  {}, 
  el("li", {}, "item 1"),
  el("li", {}, "item 2"),
  el("li", {}, "item 3"),
  el("li", {}, "item 4"),
);
```

`.flat()` has been added to prevent this from being a breaking change.

theScottyJam commented:
Thanks for the contribution!

Do you think you could
1. Since the `fragement()` utility has a similar API, do you think you could update it as well, so they can be consistent?
2. Maybe add a little something in the doc page for both `el()` and `fragment()` to explain this alternative usage pattern?

crhallberg committed:
el: spread children parameter

Spreading the children parameter opens up the opportunity for some more flexible and concise code. This can be done currently with an array, but this syntax prevents the need for an array when only one child is added.

```JS
el("p", { class: "lead" }, "text");

el(
  "ul",
  {},
  el("li", {}, "item 1"),
  el("li", {}, "item 2"),
  el("li", {}, "item 3"),
  el("li", {}, "item 4"),
);
```

`.flat()` has been added to prevent this from being a breaking change.

crhallberg committed:
Merge branch 'patch-1' of https://github.com/crhallberg/snap.js into patch-1

crhallberg commented:
Let me know if the documentation is too much or too little!

crhallberg committed:
feat: update documentation and extend to fragment.

theScottyJam commented:
Awesome, thank you for the contribution!
